repos/system_upgrade/common/actors/ipuworkflowconfig/libraries/ipuworkflowconfig.py

```diff
@@ -89,41 +40,42 @@ def get_booted_kernel():
-
-    In case the environment variable `LEAPP_DEVEL_TARGET_RELEASE` is set, the value of it will be returned.
-    """
-
-    current_version_id = get_os_release('/etc/os-release').version_id
-    target_version = upgrade_paths_map.get((current_version_id, flavour), None)
-    if not target_version:
-        # If we cannot find a particular major.minor version in the map,
-        # we fallback to pick a target version just based on a major version.
-        # This can happen for example when testing not yet released versions
-        major_version = get_major_version(current_version_id)
-        target_version = upgrade_paths_map.get((major_version, flavour), None)
-
-    return os.getenv('LEAPP_DEVEL_TARGET_RELEASE', None) or target_version
-
-
-def detect_sap_hana():
-    """
-    Detect SAP HANA based on existance of /hana/shared/*/exe/linuxx86_64/hdb/sapcontrol
+def get_os_release(path):
     """
-    if os.path.exists(HANA_BASE_PATH):
-        for entry in os.listdir(HANA_BASE_PATH):
-            # Does /hana/shared/{entry}/exe/linuxx86_64/hdb/sapcontrol exist?
-            if os.path.exists(os.path.join(HANA_BASE_PATH, entry, HANA_SAPCONTROL_PATH)):
-                return True
-    return False
+    Retrieve data about System OS release from provided file.
 
-
-def get_upgrade_flavour():
-    """
-    Returns the flavour of the upgrade for this system.
+    :return: `OSRelease` model if the file can be parsed
+    :raises: `IOError`
     """
-    if detect_sap_hana():
-        return LEAPP_UPGRADE_FLAVOUR_SAP_HANA
-    return LEAPP_UPGRADE_FLAVOUR_DEFAULT
+    try:
+        with open(path) as f:
+            data = dict(l.strip().split('=', 1) for l in f.readlines() if '=' in l)
+            return OSRelease(
+                release_id=data.get('ID', '').strip('"'),
+                name=data.get('NAME', '').strip('"'),
+                pretty_name=data.get('PRETTY_NAME', '').strip('"'),
+                version=data.get('VERSION', '').strip('"'),
+                version_id=data.get('VERSION_ID', '').strip('"'),
+                variant=data.get('VARIANT', '').strip('"') or None,
+                variant_id=data.get('VARIANT_ID', '').strip('"') or None
+            )
+    except IOError as e:
+        raise StopActorExecutionError(message='Cannot collect the system OS facts.',
+                                      details={'details': str(e)})
+
+
+def produce_ipu_config(actor):
+    flavour = os.environ.get('LEAPP_UPGRADE_PATH_FLAVOUR')
+    target_version = os.environ.get('LEAPP_UPGRADE_PATH_TARGET_RELEASE')
+    os_release = get_os_release('/etc/os-release')
+    actor.produce(IPUConfig(
+        leapp_env_vars=get_env_vars(),
+        os_release=os_release,
+        architecture=platform.machine(),
+        version=Version(
+            source=os_release.version_id,
+            target=target_version
+            ),
+        kernel=get_booted_kernel(),
+        flavour=flavour
+        ))
```

Comment:
Just small reformatting if you don't mind. But it's nitpick. Keeping on you
```suggestion
        version=Version(
            source=os_release.version_id,
            target=target_version
        ),
        kernel=get_booted_kernel(),
        flavour=flavour
    ))
```